src/main/webapp/resources/js/validation-my-booking.js

```diff
@@ -1,9 +1,37 @@
 function validateDate() {
-    var startDate = document.getElementById("from").value.toString().replace(/-/gi, "");
-    var endDate = document.getElementById("to").value.toString().replace(/-/gi, "");
-    if (endDate - startDate < 0) {
-        document.getElementById("errorDate").innerHTML = messages.date.fromBiggerThanTo;
-    } else {
-        document.getElementById("errorDate").innerHTML = "";
+    if(validateBookingsDate() == true) {
+        var startDate = document.getElementById("from").value.toString();
+        var endDate = document.getElementById("to").value.toString();
+        if (new Date(startDate) > new Date(endDate)) {
+            document.getElementById("errorDate").innerHTML = messages.date.fromBiggerThanTo;
+        } else {
+            document.getElementById("errorDate").innerHTML = "";
+        }
     }
 }
+
+function validateBookingsDate() {
+
+    var fromDateValue = $('#from').val();
+    var toDateValue = $('#to').val();
+    var regex = /^(\d{4})(\/|-)(\d{1,2})(\/|-)(\d{1,2})$/;
```

Comment:
Regexp is very insidious thing, where are we very easy will made obscure mistake. So check this twice before merging.